### PR TITLE
fix(SUP-42730): Captions not showing after enabling

### PR DESCRIPTION
**Pull request by Tzipi-kaltura**
### Description of the Changes

**Issue:**
When play the video with caption disabled, caption is not shown after enable the caption (when the caption is default)

**Fix:**
Listen to SUBTITLE_FRAG_PROCESSED (hls event) that's triggered after the captions request and parsing then hide the captions from our side.

#### Resolves [SUP-42730](https://kaltura.atlassian.net/browse/SUP-42730)

[SUP-42730]: https://kaltura.atlassian.net/browse/SUP-42730?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/hls-adapter.ts [hunk 1]
```diff
@@ -799,7 +806,11 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
   public hideTextTrack(): void {
     if (this._hls) {
       if (this._hls.subtitleTracks.length) {
-        this._hls.subtitleTrack = -1;
+        if (this._firstHideTextTrack){
```
Comment (SivanA-Kaltura) [inline src/hls-adapter.ts:809]: do you need this even if the track is not the default ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/hls-adapter.ts (hunk 1 above)
Comment (Tzipi-kaltura) [reply src/hls-adapter.ts:809]: not sure it's necessary but it's working in both cases (i did many tests) so not think need to add additional condition.

Comment (SivanA-Kaltura) [reply src/hls-adapter.ts:809]: Ok.
I think this is a good chance to refactor this function before it becomes less readable.
Something like:

public hideTextTrack(): void {
    if (!this._hls) return;

    if (!this._hls.subtitleTracks.length) {
        this.disableNativeTextTracks();
    } else if (this._firstHideTextTrack) {
        this._hls.on(Hlsjs.Events.SUBTITLE_FRAG_PROCESSED, this. _onSubtitleFragProcessed, this._hls)
    } else {
        this._hls.subtitleTrack = -1;
    }
  }

File: src/hls-adapter.ts [hunk 2]
```diff
@@ -791,6 +792,12 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     this._onTrackChanged(textTrack);
   }
 
+  private _onSubtitleFragProcessed(): void {
+    this._hls.subtitleTrack = -1;
+    this._firstHideTextTrack = false;
+    this._hls.off(Hlsjs.Events.SUBTITLE_FRAG_PROCESSED, this._onSubtitleFragProcessed, this._hls);
```
Comment (SivanA-Kaltura) [inline src/hls-adapter.ts:798]: are there any other cases where the event listener becomes irrelevant and should be removed ? like when the adapter is reset ?

File: src/hls-adapter.ts [hunk 3]
```diff
@@ -126,6 +126,7 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
   private _nativeTextTracksMap: any = {};
   private _lastLoadedFragSN: number = -1;
   private _sameFragSNLoadedCount: number = 0;
+  private _firstHideTextTrack: boolean = true;
```
Comment (SivanA-Kaltura) [inline src/hls-adapter.ts:129]: Please use a more readable name.
What does this variable mean ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/hls-adapter.ts (hunk 3 above)
Comment (Tzipi-kaltura) [reply src/hls-adapter.ts]: not sure how i can improve that, change to _isFirstSubtitleHide, the first time that captions is hided by the player. do you have any other suggestion?

File: src/hls-adapter.ts (hunk 2 above)
Comment (Tzipi-kaltura) [reply src/hls-adapter.ts:798]: add it under remove bindings that called in reset function

File: src/hls-adapter.ts (hunk 1 above)
Comment (Tzipi-kaltura) [reply src/hls-adapter.ts]: done

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**